verifying/verifying.go

```diff
@@ -50,19 +57,18 @@ func VerifyVRFNonce(nonce *uint64, m *shared.VRFNonceMetadata) error {
 
 // Verify ensures the validity of a proof in respect to its metadata.
 // It returns nil if the proof is valid or an error describing the failure, otherwise.
-func Verify(p *shared.Proof, m *shared.ProofMetadata, opts ...OptionFunc) error {
-	options := &option{
-		logger: &shared.DisabledLogger{},
-	}
+func Verify(p *shared.Proof, m *shared.ProofMetadata, cfg config.Config, opts ...OptionFunc) error {
+	options := defaultOpts()
 	for _, opt := range opts {
-		opt(options)
+		if err := opt(options); err != nil {
+			return err
+		}
 	}
-	if err := options.validate(); err != nil {
-		return err
+	if m.K1 != cfg.K1 {
+		return fmt.Errorf("invalid `K1` in proof metadata: %d != %d", m.K1, cfg.K1)
```

Comment:
Hmm, IMO it's wrong that metadata is validated separately in go-spacemesh. I think it should be moved here to keep proof verification consistent in a single place.

Reply:
This function already checks some parts of metadata (`NodeId` and `CommitmentAtxId`) - introduced in #67. 

Why would you prefer it this way? It would indicate that one can verify the proof against multiple metadata.

IMO there is no point in verifying proof and metadata separately. _The proof_ consists of a pair: `Proof` and `ProofMetadata`. Both must be correct at the same time. In fact, I don't really understand why they are separated in the first place (`NiPost` contains both)...

Reply:
I feel like it would be cleaner to validate the metadata and the proof separately. Imo the length checks for `NodeId` and `CommitmentAtx` should also be part of the `VerifyMetadata` function instead of this function.

I'm not insisting on this change I just think it would be cleaner